src/rules.c

```diff
@@ -23,11 +24,11 @@ int SchemaRuleType_Parse(const char *type_str, SchemaRuleType *type, QueryError
   if (!type_str || !strcasecmp(type_str, RULE_TYPE_HASH)) {
     *type = SchemaRuleType_Hash;
     return REDISMODULE_OK;
-  } else if (!strcasecmp(type_str, RULE_TYPE_JSON)) {
+  } else if (japi && !strcasecmp(type_str, RULE_TYPE_JSON)) {
```

Comment:
In principle, we may allow creation of JSON indexes even if `japi` is not yet available, as it may become available if rejson is loaded after index creation. If no `japi` then no notifications, but also no operations to listen to, so we're sure not to miss anything.